File: PlacementCellBackend/Controllers/CRUD/CompanyEmployeeController.cs
using Microsoft.AspNetCore.Mvc;
using PlacementCellBackend.DTOs.CompanyEmployee;
using PlacementCellBackend.Services.CRUD.Interfaces;

namespace PlacementCellBackend.Controllers.CRUD
{
    [ApiController]
    [Route("api/[controller]")]
    public class CompanyEmployeeController : ControllerBase
    {
        private readonly ICompanyEmployeeService _companyEmployeeService;

        public CompanyEmployeeController(ICompanyEmployeeService companyEmployeeService)
        {
            _companyEmployeeService = companyEmployeeService;
        }

        [HttpGet]
        public async Task<ActionResult<IEnumerable<CompanyEmployeeDto>>> GetCompanyEmployees()
        {
            var employees = await _companyEmployeeService.GetAllCompanyEmployeesAsync();
            return Ok(employees);
        }

        [HttpGet("{id}")]
        public async Task<ActionResult<CompanyEmployeeDto>> GetCompanyEmployee(string id)
        {
            var employee = await _companyEmployeeService.GetCompanyEmployeeByIdAsync(id);
            if (employee == null)
                return NotFound();
            return Ok(employee);
        }

        [HttpPost]
        public async Task<bool> PostCompanyEmployee(CompanyEmployeeCreateDto companyEmployee)
        {

            var created = await _companyEmployeeService.CreateCompanyEmployeeAsync(companyEmployee);
            return created;
        }

        [HttpPut("{id}")]
        public async Task<IActionResult> PutCompanyEmployee(string id, CompanyEmployeeCreateDto updatedCompanyEmployee)
        {


            var success = await _companyEmployeeService.UpdateCompanyEmployeeAsync(id, updatedCompanyEmployee);
            if (!success)
                return NotFound("Company employee not found.");

            return NoContent();
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> DeleteCompanyEmployee(string id)
        {
            var success = await _companyEmployeeService.DeleteCompanyEmployeeAsync(id);
            if (!success)
                return NotFound("Company employee not found.");

            return NoContent();
        }
    }
}

File: PlacementicsBackend/src/main/java/com/example/placementicsbackend/controllers/CRUD/CompanyEmployeeController.java
package com.example.placementicsbackend.controllers.CRUD;

import com.example.dto.CompanyEmployeeCreateDto;
import com.example.dto.CompanyEmployeeDto;
import com.example.service.CompanyEmployeeService;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/company-employee")
public class CompanyEmployeeController {

    private final CompanyEmployeeService companyEmployeeService;

    public CompanyEmployeeController(CompanyEmployeeService companyEmployeeService) {
        this.companyEmployeeService = companyEmployeeService;
    }

    /**
     * Get all company employees
     */
    @GetMapping
    public ResponseEntity<List<CompanyEmployeeDto>> getCompanyEmployees() {

        List<CompanyEmployeeDto> employees =
                companyEmployeeService.getAllCompanyEmployees();

        return ResponseEntity.ok(employees);
    }

    /**
     * Get company employee by Id
     */
    @GetMapping("/{id}")
    public ResponseEntity<CompanyEmployeeDto> getCompanyEmployee(
            @PathVariable String id) {

        CompanyEmployeeDto employee =
                companyEmployeeService.getCompanyEmployeeById(id);

        if (employee == null) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.ok(employee);
    }

    /**
     * Create company employee
     */
    @PostMapping
    public ResponseEntity<Boolean> postCompanyEmployee(
            @RequestBody CompanyEmployeeCreateDto companyEmployee) {

        boolean created =
                companyEmployeeService.createCompanyEmployee(companyEmployee);

        if (!created) {
            return ResponseEntity
                    .status(HttpStatus.BAD_REQUEST)
                    .body(false);
        }

        return ResponseEntity
                .status(HttpStatus.CREATED)
                .body(true);
    }

    /**
     * Update company employee
     */
    @PutMapping("/{id}")
    public ResponseEntity<Void> putCompanyEmployee(
            @PathVariable String id,
            @RequestBody CompanyEmployeeCreateDto updatedCompanyEmployee) {

        boolean success =
                companyEmployeeService.updateCompanyEmployee(
                        id,
                        updatedCompanyEmployee);

        if (!success) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.noContent().build();
    }

    /**
     * Delete company employee
     */
    @DeleteMapping("/{id}")
    public ResponseEntity<Void> deleteCompanyEmployee(
            @PathVariable String id) {

        boolean success =
                companyEmployeeService.deleteCompanyEmployee(id);

        if (!success) {
            return ResponseEntity.notFound().build();
        }

        return ResponseEntity.noContent().build();
    }
}